scss/_alerts.scss

```diff
@@ -24,83 +25,60 @@
     border-color: #d2b2b2;
   }
 
-  .flash-close {
-    float: right;
-    padding: 17px;     // bigger hit area
-    margin-top: -15px; // negative margins to account for bigger padding
-    margin-right: -15px;
-    margin-left: 20px;
-    color: inherit;
-    text-decoration: none;
-    cursor: pointer;
-    opacity: 0.6;
+  // Small flashes
+  &.flash-sm {
+    padding: 8px 16px;
 
-    &:hover {
-      opacity: 1;
+    .btn-sm {
+      margin-top: -2px;
     }
   }
 
-  p:last-child {
-    margin-bottom: 0;
+  // Add an icon
+  > .octicon {
+    float: left;
+    margin-right: 12px;
+    margin-top: 3px;
   }
 
-  // Flashes can have an action button instead of the cross.
-  .flash-action {
+  // Flashes can have an action button
+  .btn {
```

Comment:
I'm unsure if we can say that not every button in a flash message can be treated like this.